### PR TITLE
[Edge] Wait connection from edgesrc

### PR DESCRIPTION
**Pull request by gichan-jang**
Change the method of waiting for connection from polling to event.

Need group SR: https://github.com/nnstreamer/nnstreamer-edge/pull/225

**Self evaluation:**
1. Build test: [*]Passed [ ]Failed [ ]Skipped
2. Run test: [*]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: gst/edge/edge_sink.c [hunk 1]
```diff
@@ -344,36 +382,45 @@ gst_edgesink_start (GstBaseSink * basesink)
...
-      return FALSE;
+/**
+ * @brief If wait-connection is enabled, wait for connection until the connection is established or timeout occurs. Otherwise, return immediately.
+ */
+static gboolean
+_wait_connection (GstEdgeSink *sink)
+{
+  gint64 end_time;
+
+  if (!sink->wait_connection || sink->is_connected)
```
Comment (jaeyun-jung) [inline gst/edge/edge_sink.c:404]: let's avoid svace issue, removing 'sink->is_connected' here.

**Review by jaeyun-jung — APPROVED** [review]

File: gst/edge/edge_sink.c (hunk 1 above)
Comment (gichan-jang) [reply gst/edge/edge_sink.c]: Updated.

**Review by anyj0527 — APPROVED** [review]
LGTM

**Review by again4you — APPROVED** [review]
Even though this PR is not passed the CI checks, it is necessary so I will merge it forcefully.

**Pull request merged**